java/core/src/test/java/com/google/protobuf/ImportOptionTest.java

```diff
@@ -29,26 +26,15 @@ public void testImportOption() throws Exception {
     UnknownFieldSet unknownFieldsMessage = messageDescriptor.getOptions().getUnknownFields();
     UnknownFieldSet unknownFieldsField = fieldDescriptor.getOptions().getUnknownFields();
 
-    // TODO: Currently linked in options also end up in unknown fields.
-    // TODO: Exclude for open source tests once linked in options are treated
-    // differently, since `option_deps` are treated as `deps` in Bazel 7.
-    // assertThat(fileDescriptor.getOptions().getExtension(UnittestCustomOptions.fileOpt1))
-    //     .isEqualTo(1);
-    // assertThat(messageDescriptor.getOptions().getExtension(UnittestCustomOptions.messageOpt1))
-    //     .isEqualTo(2);
-    // assertThat(fieldDescriptor.getOptions().getExtension(UnittestCustomOptions.fieldOpt1))
-    //     .isEqualTo(3);
-    assertThat(unknownFieldsFile.getField(7736974).getVarintList()).containsExactly(1L);
-    assertThat(unknownFieldsMessage.getField(7739036).getVarintList()).containsExactly(2L);
-    assertThat(unknownFieldsField.getField(7740936).getFixed64List()).containsExactly(3L);
-
-    // Options from import option that are not linked in should be in unknown fields.
-    assertThat(unknownFieldsFile.getField(7736975).getVarintList()).containsExactly(1L);
-    assertThat(unknownFieldsMessage.getField(7739037).getVarintList()).containsExactly(2L);
-    assertThat(unknownFieldsField.getField(7740937).getFixed64List()).containsExactly(3L);
+    assertThat(fileDescriptor.getOptions().getExtension(UnittestCustomOptions.fileOpt1))
+        .isEqualTo(1);
+    assertThat(messageDescriptor.getOptions().getExtension(UnittestCustomOptions.messageOpt1))
+        .isEqualTo(2);
+    assertThat(fieldDescriptor.getOptions().getExtension(UnittestCustomOptions.fieldOpt1))
+        .isEqualTo(3);
 
-    assertThat(unknownFieldsFile.asMap()).hasSize(2);
-    assertThat(unknownFieldsMessage.asMap()).hasSize(2);
-    assertThat(unknownFieldsField.asMap()).hasSize(2);
+    // TODO: Since `option_deps` are treated as `deps` in Bazel 7, the unknown
```

Comment:
Nit: Would it make sense to check that unknown fields has size <= 1 instead for now? I think the comment as-is is a tad misleading since iirc option_deps being treated as deps means the unknown fields *are* empty when they should have 1 entry in it otherwise right?

Reply:
Good point, let me add you on the actual cl since it got held up by wald anyway :P